db/db.ts

```diff
@@ -310,3 +315,50 @@ export const getHomepageId = (
             AND published = TRUE`
     ).then((result) => result?.id)
 }
+
+export const getImageMetadataByFilenames = async (
+    knex: KnexReadonlyTransaction,
+    filenames: string[]
+): Promise<Record<string, ImageMetadata>> => {
+    if (filenames.length === 0) return {}
+    const rows = (await knexRaw(
```

Comment:
```suggestion
    const rows = (await knexRaw<Pick<DbEnrichedImage, "id" | "googleId" | "filename" | "defaultAlt" | "updatedAt" | "originalWidth" | "originalHeight">> (
```
I think this pattern is nicer where instead of casting to the full type in the end you pick at the point of the query

Reply:
True! I've also changed `ImageMetadata` to derive from `DbEnrichedImage` to hopefully keep things in sync a little better.